service/storage_service.hh

```diff
@@ -147,11 +146,14 @@ private:
     gms::feature_service& _feature_service;
     distributed<replica::database>& _db;
     gms::gossiper& _gossiper;
-    std::unique_ptr<service::raft_group0> _group0;
     sharded<netw::messaging_service>& _messaging;
     sharded<service::migration_manager>& _migration_manager;
     sharded<repair_service>& _repair;
     sharded<streaming::stream_manager>& _stream_manager;
+
+    // Engaged on shard 0 after `join_cluster`.
+    service::raft_group0* _group0;
```

Comment:
Should probably default-initialize it to `nullptr` (in theory no function in `storage_service` uses `_group0` before `join_cluster` is called; in practice, bugs do exists)